### ISSUE TITLE
Add option to show UI and tabs in full screen mode for Windows

### ISSUE THREAD

**Issue by MrRobot12** [brave/brave-browser#2635]
I'm using the latest developer Brave build based on Chromium on Windows 10.

When going into fullscreen mode (F11) the menu and tabs are hidden and there is no way to see them without existing out of fullscreen mode.

It should work like Firefox where the tabs and menu autohides and then appears when you move the mouse to the top of the screen.

Otherwise, what's the point of fullscreen mode if you can't actually do anything?

**Comment by shubham01** [comment]
@bbondy @MrRobot12 this also happens on MacOS 10.14.1 every time. All the ui elements including the url bar are hidden.

**Comment by shubham01** [comment]
It is happening on the stable brave version 0.58.18 @bbondy

**Comment by srirambv** [comment]
Its the same behaviour as Chrome. 

>what's the point of fullscreen mode if you can't actually do anything?

@MrRobot12 the point of fullscreen is for view only on that particular tab. I don't think there is anything else that is done on full screen. Pretty sure on FF, even if you switch tab it will exit full screen mode. Am not sure if we really need to show the tabs and file menu as doing anything there will exit from fullscreen.  From my understanding full screen is just to get extra view space for the tab and do nothing else.

**Comment by MrRobot12** [comment]
@srirambv if you switch tabs on Firefox it doesn't exit fullscreen. I believe Chrome on MacOS behaves the same way (tabs/menu autohide until mouse cursor goes to top of the screen).

The reason I use fullscreen is to avoid burn-in while using my oled as a pc monitor. I can't use Brave without it.

**Comment by srirambv** [comment]
>if you switch tabs on Firefox it doesn't exit fullscreen. 

That's neat. I wasn't aware of this feature on FF. This would be a good feature if tabs switches retained the full screen view.

We did have the show on hover option on muon but that had usability issue and hence we had static UI on top even when you went full screen so the tabs and file menu persisted even on full screen.

**Comment by bbondy** [comment]
Noticed that this option is available on Mac but not Windows.


<img width="316" alt="screen shot 2018-12-30 at 7 10 14 pm" src="https://user-images.githubusercontent.com/831718/50552862-86413d80-0c6a-11e9-8777-4a25d7b4964d.png">

If someone wants to try to see if it works on Windows:

`git grep IDC_TOGGLE_FULLSCREEN_TOOLBAR`


```
#if defined(OS_MACOSX)
    case IDC_TOGGLE_FULLSCREEN_TOOLBAR:
      chrome::ToggleFullscreenToolbar(browser_);
      break;
    case IDC_TOGGLE_JAVASCRIPT_APPLE_EVENTS: {
      chrome::ToggleJavaScriptFromAppleEventsAllowed(browser_);
      break;
    }
#endif
```

`chrome/browser/ui/browser_command_controller.cc`


Pref that's used on the mac side
```
PrefService* prefs = browser->profile()->GetPrefs();
    SetToggleState(prefs->GetBoolean(prefs::kShowFullscreenToolbar), item);
```


It's possible this is as simple as just hooking up a menu item for it.

**Comment by Brave-Matt** [comment]
Have a relevant user report: https://community.brave.com/t/ctrl-tab-is-hijacked-in-full-screen/78944/10 claiming that they're unable to use certain commands and/or switch tabs while in fullscreen mode (Windows).

@srirambv,
> the point of fullscreen is for view only on that particular tab. I don't think there is anything else that is done on full screen. Pretty sure on FF, even if you switch tab it will exit full screen mode. 

☝️ That's actually not true -- when in fullscreen, you can `ctrl + tab` to switch between tabs and in addition, you can use `Ctrl + L` to move focus to inside the address bar. When doing this, the address bar, bookmarks toolbar, and tabs all appear until focus is returned to the tab/site:
![testfs](https://user-images.githubusercontent.com/3299098/64998807-7326da00-d89a-11e9-9d24-b7a616472324.gif)


@bbondy -- Windows doesn't have the menu bar at the top as macOS does which I believe is where you're seeing this option. You can enter FS from Menu on windows.

**Other things I found:**
It seems that if an embedded player or frame has the focus, these commands (`Ctrl + L` or `Ctrl + tab`) sometimes will not work (see example in thread linked above). Additionally, **on Windows**, I'm unable to display the address bar at all using any keyboard commands (like `Ctrl + L`) or moving the mouse to the top of the screen for _any site_. 

Whereas **on macOS**, with the exception of the embedded elements limitation stated above, macOS allows me to display address bar, switch tabs, close tabs, etc without issue.

**Comment by rebron** [comment]
cc: @simonhong 

When you get a chance, can you take a look? Basically we want to be able to show/hide browser UI when a user hovers to the top. This is Windows only as it's working on macOS. I noticed this feature working properly in Whale on Windows.

**Comment by simonhong** [comment]
@rebron There is no hover on top feature on macOS and Windows.
And it's same on brave and Chrome.
The only difference between macOS and Windows is user can make browser ui visible in full screen mode.

In Whale, hover on top feature is available only on Windows. IIRC, it's customized immersive mode of chromeos feature.

**Comment by rebron** [comment]
@simonhong There is a hover on top (or move mouse to top of the screen) on both macOS and Windows, just different behavior as you described [ (1) 'x' to exit Full screen mode in Windows, (2) show browser UI in macOS if 'Allow show toolbar in Full Screen mode' is hidden] and we are matching Chrome behavior.

The feature request is on Windows, make browser ui visible in full screen mode when a user hovers to the top. This should match the same functionality as in Whale and in Firefox where placing mouse to the top screen shows browser ui.

https://github.com/brave/brave-browser/issues/2635#issuecomment-450597441 we don't have this pref for windows or a place to expose/toggle it. Right click context menu could work.

**Comment by simonhong** [comment]
@rebron We need to clarify about `browser UI` when mouse is hovered on top.
On macOS, `browser UI` means browser menu that sliding down when hovered on top.
On Windows, `browser UI` means `X` button UI to exit Full screen.

And what we want is showing `browser UI(toolbar, omnibox and bookmarks)` with sliding animation when hovered on top only for Windows. right?

**Comment by rebron** [comment]
@simonhong That's right. For Windows sliding animation when mouse is hovered on top show browser UI (tabs, navbar/omnibox, and bookmarks)

**Comment by SHHSSH** [comment]
Anyone update on the functionality of Ctrl+L in fullscreen mode? It's rather nuisance-some to have to revert to maximized mode just to hotkey to the address bar....

**Comment by samuelmattjohnston** [comment]
Any updates here? this issue has made it way into Linux. Swapping to another browser since the Linux build has no alternatives for dynamic url bar in fullscreen. Further, this has completely broken video fullscreen playback (have to refreshpage or maximize/minimize multiple times quickly to get it to revert to non fullscreen mode).

**Comment by iurimattos** [comment]
This is meant to be a joke answer. Try this extension: https://chrome.google.com/webstore/detail/tab-url-manager/egiemoacchfofdhhlfhkdcacgaopncmi/related

how to use: https://support.brave.com/hc/en-us/articles/360017909112-How-can-I-add-extensions-to-Brave-

**Comment by MrRobot12** [comment]
Almost 2 years later and no updates from Brave.

I guess this will never be implemented. 😑

**Comment by jochienabuurs** [comment]
When using Mac OS, this is also still an issue. Using full screen or split-screen mode causes all tabs to disappear.
I'm a webdeveloper and use split screen quite often to show both my editor and the browser. When using Brave, all tabs are gone, which quite frankly makes Brave unusable in split screen mode

**Comment by dserodio** [comment]
That's weird, I'm seeing the opposite behaviour with Brave 1.18.78 on macOS Catalina (10.15.7), and I'd like it be more like Chrome.
![image](https://user-images.githubusercontent.com/233589/104071791-a8baa980-51e8-11eb-8989-0697c59fb78d.png)

**Comment by bbley1** [comment]
I dont see any show toolbar option in Windows version.  and OP is asking for something like Edge's ShyUI feature or Firefox's Fullscreen Mode.

heres a demonstration :

https://techcommunity.microsoft.com/t5/discussions/finally-here-the-real-full-screen-experience-in-edge-insider/td-p/1011888

**Comment by SHHSSH** [comment]
> 
> 
> This should be closed. There is `View -> Always Show Toolbar in Full Screen` which seems like it covers both needs expressed here.

I understand that in regards to OP's post, however I still think Brave lacks major functionality in regards to my post too.



> 
> 
> Any update on the functionality of Ctrl+L in fullscreen mode? It's rather tedious to have to revert to maximized mode just to hotkey to the address bar....

Besides, who wants to see the Toolbar when they're in Full Screen?

Eh... I doubt this will be addressed anytime soon anyway, since Chrome still doesn't even offer Ctrl + L functionality when in Full Screen to navigate to the address bar.

**Comment by alturkovic** [comment]
> I dont see any show toolbar option in Windows version. and OP is asking for something like Edge's ShyUI feature or Firefox's Fullscreen Mode.
> 
> heres a demonstration :
> 
> https://techcommunity.microsoft.com/t5/discussions/finally-here-the-real-full-screen-experience-in-edge-insider/td-p/1011888

You are right, I was too quick to speak. It works just like expected on MacOS, and I assumed (wrongly) that it would work the same on Windows 10, but it doesn't.

My bad.

**Comment by bradleyhh** [comment]
> This is meant to be a joke answer but try this extension: https://chrome.google.com/webstore/detail/tab-url-manager/egiemoacchfofdhhlfhkdcacgaopncmi/related
> 
> how to use: https://support.brave.com/hc/en-us/articles/360017909112-How-can-I-add-extensions-to-Brave-

Temp fix but still cannot see bookmarks

**Comment by bradleyhh** [comment]
>**on Windows**, I'm unable to display the address bar at all using any keyboard commands (like `Ctrl + L`) or moving the mouse to the top of the screen for _any site_.

Any fix to this?

**Comment by rossunger** [comment]
+1

**Comment by XeroraYT** [comment]
2023 and still no implementation from what I've been able to find.

**Comment by aztral** [comment]
This feature and "open tabs next to current tab" are the two primary features preventing me from fully switching to brave browser.

As a alternative for the missing adress bar I just use a plugin i can open with a hotkey to create a new browsing tab with a search query. 

Its weird that CTRL + T doesnt even allow you to type blindly in the adress bar. Perhaps create a adress bar widget in the brave home page screen...

**Comment by Brave-Matt** [comment]
Another +1 here -- @rebron can we get a status update on this? Looks like it hasn't been looked at since last year even though set as `P3`.
https://community.brave.app/t/address-bar-and-other-controls-to-full-screen-mode/644352/9

**Comment by SiLenCe-WB** [comment]
Any news on this feature? This would be a great thing to have in this day and age of OLED monitors.

**Comment by SiLenCe-WB** [comment]
Anybody listening?

**Comment by GreyTen1** [comment]
.

**Comment by patakihara** [comment]
+1

**Comment by Brave-Matt** [comment]
+1 https://community.brave.app/t/auto-hide-browser-toolbar-in-fullscreen-on-windows/657478/2

**Comment by KapitanHash** [comment]
+1
